sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonJsonSerializer.java

```diff
@@ -34,6 +37,46 @@
  * Jackson based implementation of the {@link JsonSerializer} and {@link MemberNameConverter} interfaces.
  */
 public final class JacksonJsonSerializer implements JsonSerializer, MemberNameConverter {
+    private static final String ACCESSOR_NAMING_STRATEGY =
+        "com.fasterxml.jackson.databind.introspect.AccessorNamingStrategy";
+    private static final String ACCESSOR_NAMING_STRATEGY_PROVIDER = ACCESSOR_NAMING_STRATEGY + ".Provider";
+    private static final MethodHandle GET_ACCESSOR_NAMING;
+    private static final MethodHandle FOR_POJO;
+    private static final MethodHandle FIND_NAME_FOR_IS_GETTER;
+    private static final MethodHandle FIND_NAME_FOR_REGULAR_GETTER;
+    private static final boolean USE_REFLECTION_FOR_MEMBER_NAME;
+
+    static {
+        MethodHandles.Lookup publicLookup = MethodHandles.publicLookup();
+
+        MethodHandle getAccessorNaming = null;
+        MethodHandle forPojo = null;
+        MethodHandle findNameForIsGetter = null;
+        MethodHandle findNameForRegularGetter = null;
+        boolean useReflectionForMemberName = false;
+
+        try {
+            Class<?> accessorNamingStrategyProviderClass = Class.forName(ACCESSOR_NAMING_STRATEGY_PROVIDER);
+            Class<?> accessorNamingStrategyClass = Class.forName(ACCESSOR_NAMING_STRATEGY);
+            getAccessorNaming = publicLookup.findVirtual(MapperConfig.class, "getAccessorNaming",
+                MethodType.methodType(accessorNamingStrategyProviderClass));
+            forPojo = publicLookup.findVirtual(accessorNamingStrategyProviderClass, "forPOJO",
+                MethodType.methodType(accessorNamingStrategyClass, MapperConfig.class, AnnotatedClass.class));
+            findNameForIsGetter = publicLookup.findVirtual(accessorNamingStrategyClass, "findNameForIsGetter",
+                MethodType.methodType(String.class, AnnotatedMethod.class, String.class));
+            findNameForRegularGetter = publicLookup.findVirtual(accessorNamingStrategyClass, "findNameForRegularGetter",
+                MethodType.methodType(String.class, AnnotatedMethod.class, String.class));
+            useReflectionForMemberName = true;
+        } catch (Throwable ignored) {
+        }
+
+        GET_ACCESSOR_NAMING = getAccessorNaming;
+        FOR_POJO = forPojo;
+        FIND_NAME_FOR_IS_GETTER = findNameForIsGetter;
+        FIND_NAME_FOR_REGULAR_GETTER = findNameForRegularGetter;
+        USE_REFLECTION_FOR_MEMBER_NAME = useReflectionForMemberName;
```

Comment:
If the above `try` block throws, some of these could be null. It would be better to throw instead of ignoring the exception and to get into an inconsistent state here.

Reply:
Given that this is in a static constructor this could result in an application failing to start, and this would happen if Spring resolves to 2.11 as this is trying to fix. Instead, would it better to validate that all constants are non-null? There is a flag that only gets set to true if no exceptions are thrown to use these constants.

Reply:
Additionally, I'm not certain if `MethodHandle` lookups can return null, I believe they throw if a handle isn't found (probably for this exact reason, to not get into inconsistent states).

Reply:
Yeah, it seems like MethodHandle will never return null. So, just using the boolean at the end makes sense to ensure everything is initialized. Maybe just log the exception in the catch block above.